addons/website_sale/models/sale_order.py

```diff
@@ -426,13 +358,100 @@ def _cart_find_product_line(
 
         return filtered_sol
 
+    def _cart_update_line_quantity(self, line_id:int, quantity:int|float, **kwargs):
+        """Update the quantity of a given line of the cart.
+
+        :param int line_id: line id, as a `sale.order.line` id.
+        :params float quantity: the updated quantity of the line.
+        :param dict kwargs: Additional parameters given to deeper method calls.
+        :return: values used by the cart service to give feedback to the customer.
+        :rtype: dict
+        """
+        self.ensure_one()
+        self = self.with_company(self.company_id)
+
+        if not (order_line := self.order_line.filtered(lambda sol: sol.id == line_id)):
+            raise UserError(_("This line doesn't belong to your order."))
+
+        if quantity > 0:
+            quantity, warning = self._verify_updated_quantity(
+                order_line,
+                order_line.product_id.id,
+                quantity,
+                **kwargs,
+            )
+        else:
+            # If the line will be removed anyway, there is no need to verify
+            # the requested quantity update.
+            warning = ''
+
+        order_line = self._cart_update_order_line(order_line, quantity, **kwargs)
+        self._verify_cart_after_update(order_line)
+
+        return {
+            'line_id': order_line.id,
+            'quantity': quantity,
+            'warning': warning,
+        }
+
     # hook to be overridden
     def _verify_updated_quantity(self, order_line, product_id, new_qty, **kwargs):
         return new_qty, ''
 
+    def _cart_update_order_line(self, order_line, quantity, **kwargs):
+        self.ensure_one()
+        order_line.ensure_one()
+
+        if quantity <= 0:
+            # Remove zero or negative lines
+            order_line.unlink()
+            return self.env['sale.order.line']
+
+        # Update existing line
+        update_values = self._prepare_order_line_update_values(order_line, quantity, **kwargs)
+        if update_values:
+            order_line.write(update_values)
+
+            # If the line is a combo product line, and it already has combo items, we need to update
+            # the combo item quantities as well.
+            if (
+                order_line.product_type == 'combo'
+                and order_line.linked_line_ids
+                and 'product_uom_qty' in update_values
+            ):
+                for linked_line_id in order_line.linked_line_ids:
+                    if quantity != linked_line_id.product_uom_qty:
```

Comment:
Can't we update the quantity on all the line at the same time? Or maybe it isn't worth it? 

BTW `_verify_updated_quantity` is `self.ensure_one()` in `website_sale_collect` which is really weird IMO, an override shouldn't decide if the method is ensure_one or not